js/main.js

```diff
@@ -100,22 +111,55 @@ function viewSwapping(view) {
   if (view === 'edit-profile') {
     $edit.className = 'edit';
     $profile.className = 'profile hidden';
+
+    if (data.profile.avatarUrl === null || data.profile.avatarUrl === '') {
+      $avatarEditImage.setAttribute('src', 'images/placeholder-image-square.jpg');
+    } else {
+      $avatarEditImage.setAttribute('src', data.profile.avatarUrl);
+    }
+
+    $avatarURL.setAttribute('value', data.profile.avatarUrl);
+    $userName.setAttribute('value', data.profile.username);
+    $fullName.setAttribute('value', data.profile.fullName);
+    $location.setAttribute('value', data.profile.location);
+    $bio.textContent = data.profile.bio;
   }
 
   if (view === 'profile') {
     $profile.className = 'profile';
     $edit.className = 'edit hidden';
+
+    var $container = document.querySelector('.container');
+    $profile.removeChild($container);
+
     var newDomtree = domTreeCreation(data.profile);
     $profile.appendChild(newDomtree);
   }
 
 }
 
-// Allow users to see their created profile
+// New User Screen VS Returning User Screen
 document.addEventListener('DOMContentLoaded', function (event) {
   if (data.profile.username === '' || data.profile.username === null) {
     viewSwapping('edit-profile');
   } else {
     viewSwapping('profile');
   }
 });
+
+document.addEventListener('click', function (event) {
+  var $editButtonContainer = document.querySelector('.editButton');
+  if (event.target === $editButtonContainer) {
+    viewSwapping('edit-profile');
+  }
+});
+
+document.addEventListener('click', function (event) {
+  var $navbarProfile = document.querySelector('.navbarProfile');
+  if (event.target === $navbarProfile) {
+    viewSwapping('profile');
```

Comment:
This should be in a condition below the check if username is empty. As of right now, if their username is empty, the user is being navigate to the profile and then immediately to edit profile. It wouldn't be noticeable to a user most likely, but it is extra work for the browser.